### DISCUSSION TITLE
Android mocking framework of choice

### DISCUSSION THREAD

**Discussion by fredells** [MobileNativeFoundation/discussions#108] [General]
**What are you using? How is it working for you? What are the gaps?**

Context:
Previously I worked at Shopify and we used Mockito almost exclusively. Our codebase was pretty much full kotlin and we used coroutines very sparsely. It has a bit of a learning curve especially around readability. But it got the job done effectively.

Currently I am working at Reddit and we are evaluating mocking frameworks as we seek to improve our unit test coverage. MockK looks like a strong option. Especially since our codebase leans heavily into coroutines.

Does anyone have experience with MockK vs Mockito? Or any of the other options like PowerMock, EasyMock,  etc.

**Comment by simtop** [comment]
Hello I used MockK extensively, and in my opinion it's more helpful than mockito when working with a Kotlin codebase. 
The things I like about MockK is the styling, the api usage is more of the kotlin style than mockito. Also it has support for coroutines like coEvery, coVerify... You can mock final classes, static objects and extension functions.

The hardest part for me was to convince my team to use MockK because they were used to mockito, but the syntax is not that different, so it's easy and fast to get used to MockK.
